From a13a287723aa5cb6c60a260eb97b4f1d91d53599 Mon Sep 17 00:00:00 2001
From: Tim Blechmann <tim.blechmann@qt.io>
Date: Sun, 14 Sep 2025 12:39:43 +0800
Subject: [PATCH 084/553] expected: do not include qconfig.h

qconfig.h does not have inclusion guards, so it should no be included
directly: it can break code as qcompilerdetection.h undef-ines its
symbols.

Fixes: QTBUG-140149
Task-number: QTBUG-140150
Change-Id: Ic655a87ebeccd3cbc66286fd92279da8ca11cad3
Reviewed-by: Volker Hilsheimer <volker.hilsheimer@qt.io>
(cherry picked from commit 901ccf760891cc3aa1adf2f6c8105c6417f4783a)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../0001-tl-expected-disable-exceptions-with-qt-s-macros.patch  | 2 +-
 src/corelib/global/qexpected_p.h                                | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/global/patches/tlexpected/0001-tl-expected-disable-exceptions-with-qt-s-macros.patch b/src/corelib/global/patches/tlexpected/0001-tl-expected-disable-exceptions-with-qt-s-macros.patch
index 19e889d5d5c..b3c120f245c 100644
--- a/src/corelib/global/patches/tlexpected/0001-tl-expected-disable-exceptions-with-qt-s-macros.patch
+++ b/src/corelib/global/patches/tlexpected/0001-tl-expected-disable-exceptions-with-qt-s-macros.patch
@@ -15,8 +15,8 @@ index a54aea1f7d7..45c6196c947 100644
 @@ -34,6 +34,7 @@
  #include <QtCore/private/qglobal_p.h>
  #include <QtCore/qassert.h>
++#include <QtCore/qtconfiginclude.h>
  #include <QtCore/qtconfigmacros.h>
-+#include <QtCore/qconfig.h>
  
  #include <exception>
  #include <functional>
diff --git a/src/corelib/global/qexpected_p.h b/src/corelib/global/qexpected_p.h
index 4265435d2f7..24ea5be1e5e 100644
--- a/src/corelib/global/qexpected_p.h
+++ b/src/corelib/global/qexpected_p.h
@@ -36,8 +36,8 @@
 
 #include <QtCore/private/qglobal_p.h>
 #include <QtCore/qassert.h>
+#include <QtCore/qtconfiginclude.h>
 #include <QtCore/qtconfigmacros.h>
-#include <QtCore/qconfig.h>
 
 #include <exception>
 #include <functional>
-- 
2.51.2

